src/ape_ethereum/provider.py

```diff
@@ -917,6 +917,11 @@ def send_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
             # a VM error.
             receipt.raise_for_status()
 
+        # TODO: Optional configuration?
+        if txn.data:
+            # Look for and print any contract logging
+            log_print(receipt)
```

Comment:
I'm not really big on the placement of this because it requires implementation in all relevant providers.  In fact, there's a bit of redundant code between all the providers I checked.  For instance, the history addition.

https://github.com/ApeWorX/ape-foundry/blob/0b6b55f370c6bb1d460c7100a07b5b718b1496d7/ape_foundry/provider.py#L576

I'm wondering of there should be a `post_transaction(tx, receipt)` method on `Web3Provider` that gets calls on all implementations.  Or maybe there's a better idea?  Just seems annoyingly redundant to have to call this same function from every provider that implements `send_transaction()`.

Reply:
I added the hook.  That way subclassed providers will also have this functionality.